File: Práctica 1/Ruby/shield.rb
# encoding: UTF-8

module Irrgarten

    class Shield

        def initialize(protection, uses)

            @protection = protection
            @uses = uses

        end

        def protect

            return 0 if @uses <= 0

            @uses -= 1
            return @protection

        end

        def discard
            return Dice.discard_element(@uses)
        end

        def to_s
            return "S[" + @protection.to_s + ", " + @uses.to_s + "]"
        end

    end

end

File: Práctica 1/Java/IrrgartenP1/src/irrgarten/Shield.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author aloncraftmc
 */
public class Shield {
    
    private float protection;
    private int uses;
    
    public Shield(float protection, int uses){
        
        this.protection = protection;
        this.uses = uses;
        
    }
    
    public float protect(){
        
        if(this.uses <= 0)  return 0;
        
        this.uses--;
        return this.protection;
        
    }
    
    public boolean discard(){
        return Dice.discardElement(this.uses);
    }
    
    public String toString(){
        return "S[" + this.protection + ", " + this.uses + "]";
    }
    
}
